add_book.php

```diff
@@ -0,0 +1,34 @@
+<?php
+include 'database.php';
```

Comment:
Observations: The current implementation directly uses raw SQL queries with user input, which is vulnerable to SQL injection attacks. Additionally, the code is procedural and lacks separation of concerns.

Recommendations:
- **Use Prepared Statements**: Utilize prepared statements to prevent SQL injection. This can be done using either the `mysqli` or `PDO` extension.
- **Refactor to Object-Oriented Design**: Implement a `BookRepository` class to handle database operations. This class should use dependency injection to receive a database connection.

Example:
```php
class BookRepository {
    private $connection;

    public function __construct($connection) {
        $this->connection = $connection;
    }

    public function addBook($title, $author, $year) {
        $stmt = $this->connection->prepare("INSERT INTO books (title, author, year) VALUES (?, ?, ?)");
        $stmt->bind_param("ssi", $title, $author, $year);
        return $stmt->execute();
    }
}
```

- **Refactor `add_book.php`**: Use the `BookRepository` to handle the book addition logic.
```php
include 'database.php';
include 'BookRepository.php';

$bookRepo = new BookRepository($connection);

if ($_SERVER['REQUEST_METHOD'] == 'POST') {
    $title = $_POST['title'];
    $author = $_POST['author'];
    $year = $_POST['year'];

    if ($bookRepo->addBook($title, $author, $year)) {
        echo "Book added successfully!";
    } else {
        echo "Error: " . $connection->error;
    }
}
```